### PR TITLE
Add MergeOperators for LeaderEvent and LeaderEventsHistory

### PR DESCRIPTION
**Pull request by gopalrajpurohit**
1. Add SingleMergeOperator to merge a new LeaderEvent into an existing LeaderEventsHistory object.
2. Add BatchMergeOperator to merge a list of LeaderEvent into an existing LeaderEventsHistory object.
3. Add Descending Order sorting comparator for maintaining the descending order of events w.r.t. timestamp after merge is completed.

### CODE REVIEW

**Comment by gopalrajpurohit** [comment]
First part of https://github.com/pinterest/rocksplicator/pull/423

File: cluster_management/src/main/java/com/pinterest/rocksplicator/eventstore/BatchLeaderEventMergeOperator.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    implements BatchMergeOperator<LeaderEventsHistory> {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(BatchLeaderEventMergeOperator.class);
+  final String resourcName;
+  final String partitionName;
+  final Optional<Integer> maxEventsToKeep;
+
+  BatchLeaderEventMergeOperator(
+      final String resourcName,
```
Comment (rajathprasad) [inline cluster_management/src/main/java/com/pinterest/rocksplicator/eventstore/BatchLeaderEventMergeOperator.java:42]: Can we rename this to `resourceName`?

**Review by rajathprasad — APPROVED** [review]
lgtm

File: cluster_management/src/main/java/com/pinterest/rocksplicator/eventstore/BatchLeaderEventMergeOperator.java (hunk 1 above)
Comment (gopalrajpurohit) [reply cluster_management/src/main/java/com/pinterest/rocksplicator/eventstore/BatchLeaderEventMergeOperator.java:42]: already done.

**Pull request merged**